Raven.Database/Actions/DocumentActions.cs

```diff
@@ -273,6 +273,8 @@ public int BulkInsert(BulkInsertOptions options, IEnumerable<IEnumerable<JsonDoc
                     cts.Token.ThrowIfCancellationRequested();
 
                     var docsToInsert = docs.ToArray();
+                    docsToInsert.ForEach(document => document.Key = string.IsNullOrWhiteSpace(document.Key) ? Guid.NewGuid().ToString() : document.Key.Trim());
```

Comment:
IMO the Guid should be removed because:
a) http://ravendb.net/docs/article-page/3.5/Csharp/client-api/bulk-insert/how-to-work-with-bulk-insert-operation#limitations
"Entity Id must be provided by the client. The client by default will use the HiLo generator in order to generate the Id."
b) https://github.com/ravendb/ravendb/blob/v3.5/Raven.Client.Lightweight/Document/RemoteBulkInsertOperation.cs#L285

so you will insert a dead code.

The best option (I think) will be to change
https://github.com/ravendb/ravendb/blob/v3.5/Raven.Database/Actions/DocumentActions.cs#L292
to IsNullOrWhiteSpace and normalize in other case